.github/workflows/check-typescript.yml

```
@@ -0,0 +1,9 @@
+name: Check for TypeScript errors
```

```suggestion
name: Test
```
Maybe we can include the test suite along with the type checking in the future, or maybe we can always keep them separate so they run in parallel? Just some ideas

---

Let's keep them separate for now. We can also separate running tests into its own action down the line instead of having it be part of the staging action.